### PR TITLE
feat: concepts

### LINKED ISSUE #4: feat(section-2): Concepts
### Background
At the start of the documentation, after the landing page, we introduce _Concepts_. This is from a high level perspective.

### Details

This section is to be owned by Marketing, and cross-checked by Engineering for (sanity checks / veracity only).

### Acceptance criteria

- [x] Introduction
- [x] Batch Auctions

**NOTE**: Update the acceptance criteria above if the documentation structure changes to reflect such.

### PR DESCRIPTION
**Pull request by avsavsavs**
# Description

This PR adds Introduction and Concepts documentation.

# Changes

- [x] Add CoW DAO docs introduction
- [x] Add CoW Protocol concepts & product features

## Related Issues

Fixes #4

### CODE REVIEW

**Comment by avsavsavs** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by mfw78** [comment]
FYI, standby for rebasing this PR.

File: docs/cow-protocol/README.mdx [hunk 1]
```diff
@@ -1 +1,11 @@
-# CoW Protocol (Batch Auctions)
\ No newline at end of file
+# CoW Protocol (Batch Auctions)
+
+CoW Protocol matches users' intents via batch auctions from a variety of on-chain liquidity sources.
+
+When matching intents, CoW Protocol first seeks a _Coincidence of Wants_ (CoW) within the existing batch to offer an even better price than any pool/market maker can. If no CoW is found, CoW Protocol then seeks the best price for a trade from available on-chain liquidity.
```
Comment (fleupold) [inline docs/cow-protocol/README.mdx]:
```suggestion
When matching intents, CoW Protocol first seeks a _Coincidence of Wants_ (CoW) within the existing batch to offer an even better price than any pool/market maker can. If no CoW is found, CoW Protocol solvers then seeks the best price for a trade from available on-chain liquidity.
```

File: docs/cow-protocol/concepts/batch-auctions/better-execution-prices.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+---
+sidebar_position: 3
+---
+
+# Better Execution Prices
+
+Batch auctions are also designed to offer a better price over traditional CFMMs. Liquidity fragmentation poses a big problem for the DeFi space, forcing users to constantly check liquidity pools against each other to find the best prices. Batch auctions help solve liquidity fragmentation by matching orders peer-to-peer through Coincidence of Wants (CoWs), thus allowing them to share liquidity. If orders are not fully filled by CoWs, the protocol acts like a more traditional aggregator and finds on-chain liquidity to complete the order. In essence, batch auctions turn the entire protocol into a giant barter economy where users can either trade directly against each other.
```
Comment (fleupold) [inline docs/cow-protocol/concepts/batch-auctions/better-execution-prices.md]: "either" without an or at the end of the sentence

File: docs/cow-protocol/concepts/batch-auctions/how-batch-auctions-work.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
+---
+sidebar_position: 5
+---
+
+# How batch auctions work
+
+Given that Ethereum settles orders in discrete blocks every 12 seconds, the CEX-native method of orderbook trading may not be the optimal mechanism for settling DeFi trades. Due to the nature of AMMs, which settle trades continuously, the order in which a transaction executes determines how good of a deal a trader is getting. Batch auctions are a DeFi-native solution that provides fair pricing for all transactions within an Ethereum block. 
+
+Ethereum has been friendly to batching mechanisms for a while. With the introduction of [PBS](https://ethereum.org/nl/roadmap/pbs/), Ethereum broke from a single-auction model into a two-sided marketplace. Builders take part in an initial auction to determine what transactions will be included in a block, and proposers take part in a second auction to decide which block will ultimately be committed to the chain next. So users are already, in fact, participating in “batch auction”-style transaction settlement. CoW Protocol simply pulls this mechanism forward to the order creation stage.
```
Comment (fleupold) [inline docs/cow-protocol/concepts/batch-auctions/how-batch-auctions-work.md]: PBS maximises payout for the validator (extract value from users), CoW protocol focusses on giving it back, EBBO not actually enforced pre-settlement

File: docs/cow-protocol/concepts/introduction/flow-of-an-order.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+2. The protocol includes the user’s intent to trade in the CoW Protocol orderbook
+3. The Auctioneer gathers multiple intent orders and groups them into a batch auction — meaning they will all settle on-chain together
+4. Solvers receive the batch _instance_ and begin finding the most optimal execution path for the various orders. The solvers compete with each other to offer the highest surplus for the orders and win the right to settle the batch on-chain
+5. Once the batch auction time window is over, the auctioneer ranks all the received _solutions_ and selects the winning solver. The winning solver executes the orders on behalf of the users and settles the batch on-chain 
+
+Once the winning solver executes the batch’s orders, users receive their tokens. This delegated trading module ensures users (including consumers, bots, smart contracts, and DAOs) always get the best prices for every trade. Letting solvers handle the heavy lifting means users don’t have to worry about gas prices, finding the best liquidity pool, or setting the optimal slippage. Solvers are also experts at avoiding MEV so users can rest assured they are protected from price exploitation.
```
Comment (fleupold) [inline docs/cow-protocol/concepts/introduction/flow-of-an-order.md]: Solvers are not just experts but also bear the risk of MEV

File: docs/cow-protocol/concepts/introduction/intents-as-signed-orders.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    * This optimization also bypasses LP fees and reduces the gas necessary for trading, since users don't need to interact with non-CoW Protocol smart contracts, but also allows the trade itself to happen by re-aggregating on-chain liquidity that otherwise was scattered all around
+* **Directional CoWs, (AKA batching trades within the batch auction):** Often, a batch auction contains multiple intents from different users who all want to perform the same trade. Instead of sending their trades individually one by one, solvers can batch all of the intents together and execute them as a single trade on-chain
+    * Although a directional CoW goes on-chain, this optimization reduces gas fees, as users trades only interact with AMM smart contracts once
+
+From a technical standpoint, CoW Protocol relies on intents as a core part of the delegated trading execution model. In this model, solvers, rather than users, are the ones in charge of actually settling the trades on-chain. The delegated trading model has several benefits including:
+
```
Comment (fleupold) [inline docs/cow-protocol/concepts/introduction/intents-as-signed-orders.md:22]: Mention pre and post trade interactions/hooks ?

File: docs/cow-protocol/concepts/product-features/builder-friendly/milkman.md [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+sidebar_position: 4
+---
+
+# Milkman orders
+
+Through Milkman, instead of specifying a ‘min out’ to receive, you specify an **on-chain** data source to pull the market price of your trading pair. 
+
+For example, if you’re swapping `ETH` / `USDC`, you can supply the `ETH` / `USDC` Chainlink price feed. Milkman will automatically adjust the minimum price that you should receive, giving you better on-chain price guarantees and reducing the chances of placing orders with stale prices.
+
+Let’s say you’re looking to sell 1,000 `ETH` and you want to make sure to get the most accurate price for your trade, even if it’s far out in the future. By sending your order through Milkman, you can rely on a price oracle to guarantee that you receive a price that is at least 99.5% (or another custom percentage) of the oracle price at any moment in time.
```
Comment (fleupold) [inline docs/cow-protocol/concepts/product-features/builder-friendly/milkman.md]: Maybe add an example of what problem this solves (better surplus protection than EBBO for large orders)

File: docs/cow-protocol/concepts/product-features/cow-hooks.md [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+
+**"Pre-hooks"**: These are hooks that execute at the start of the settlement, before the swap. Note that these are executed even before an order's signature is checked or sell tokens are transferred out of the user's account. This allows pre-hooks to be used to "set up" an order. For example:
+
+* Using a Safe and executing a `SignMessageLib` transaction in order to pay for signing an on-chain order if, and only if, it gets executed
+* Setting the required `ERC-20` approvals to the CoW Protocol using an `EIP-2612` permit, so approvals are only set if the order executes
+* Unstaking tokens just-in-time for trading
+* Claiming an airdrop right before dumping
+
+**"Post-hooks"**: These are hooks that execute at the end of a settlement, after the swap completes. This means that the trade proceeds have already been transferred to the `receiver` address and can be used by the post-hooks. For example:
+
```
Comment (fleupold) [inline docs/cow-protocol/concepts/product-features/cow-hooks.md:17]: Mention what happens for partial fills (which hooks are executed on each vs first fill)

File: docs/cow-protocol/create/cowswap-ui/limit.mdx [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+
+## Place a limit order
+
+Let's assume the example that we want to sell our `COW` that we purchased previously. We can set the limit price at which we want to sell, and the duration over which the limit order is valid:
+
+![Limit orders](/img/cowswap/limit_place_order.png)
+
+:::caution
+
+Fees for limit orders are taken from the surplus CoW Protocol generates for your order - that is, by executing your order at a better price than your limit price. CoW Protocol is capable of finding much better prices than are otherwise available due to its unique batch auction mechanism; in situations where it can't find a better price, it will wait until the market price is sufficient to cover both your limit price and the network fees. This means that your order may not execute exactly when the market price hits your limit price.
```
Comment (fleupold) [inline docs/cow-protocol/create/cowswap-ui/limit.mdx]: Should probably elaborate more on this fee mechanism (maybe with an example) as it's a very frequent cause for confusion.

File: docs/cow-protocol/create/cowswap-ui/limit.mdx [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+:::
+
+At this point, CoW Swap will prompt you to review your limit order:
+
+![Review limit order](/img/cowswap/limit_review_order.png)
+
+:::tip
+
+By default, CoW Swap sets your limit order to _partially fillable_. This allows your order to fill gradually as liquidity comes available. Partially fillable orders usually complete faster than _fill or kill_ orders and also provide better prices, so they are the default mechanism used to execute limit orders. You can turn off partially fillable limit orders and revert back to _fill or kill_ through the swap panel settings.
+
```
Comment (fleupold) [inline docs/cow-protocol/create/cowswap-ui/limit.mdx]: Maybe mention that partially fillable orders may leave you with some dust (ie not a 100% fill)

File: docs/cow-protocol/create/cowswap-ui/native.mdx [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+
+After selecting your tokens, CoW Swap will prompt you to confirm the swap. This will be a single on-chain transaction that will wrap your native tokens (if required) and place the swap intent on CoW Protocol.
+
+![Confirm Native tokens](/img/cowswap/native_confirm.png)
+
+<EthFlowTip />
+
+Once you have confirmed the swap, your wallet will prompt you to the on-chain transaction interacting with Eth-flow. On completion of this transaction, your swap intent will be placed on CoW Protocol:
+
+![Eth-flow Transaction](/img/cowswap/native_tx.png)
```
Comment (fleupold) [inline docs/cow-protocol/create/cowswap-ui/native.mdx]: Maybe mention that ETH will leave your wallet (before the buy token arrives) and also explain the refund mechanism that should get automatically triggered if your order expires.

File: docs/cow-protocol/create/cowswap-ui/swap.mdx [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+
+- Not having to approve multiple times
+- Save on gas costs
+
+<TokenApprovalWarning token="WETH" spender="GPv2VaultRelayer" />
+
+Once the `approve` transaction is confirmed, you can proceed to swap!
+
+![GPv2VaultRelayer approved](/img/cowswap/swap_approved.png)
+
```
Comment (fleupold) [inline docs/cow-protocol/create/cowswap-ui/swap.mdx]: Describe and mention slippage tolerance setting?

File: docs/cow-protocol/reference/contracts/core/README.mdx [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+1. User funds are **NOT** accessible _except_ for settling an order authorized by the user
+2. A trade **MUST** be executed at the _limit price or better_ as specified in the order
+3. Once an order is fulfilled, it **CANNOT** be traded again
+
+## Deployments
+
+:::tip
+
+CoW Protocol contracts are deployed using deterministic addresses. This means that the same contract code will always be deployed to the same address. This is done by using the `CREATE2` opcode.
+
```
Comment (fleupold) [inline docs/cow-protocol/reference/contracts/core/README.mdx:34]: Maybe mention that domain separator is different on each chain though (which has been source of confusion in the past)

File: docs/cow-protocol/reference/contracts/core/vault_relayer.md [hunk 13]
```diff
@@ -0,0 +1,99 @@
...
+# GPv2VaultRelayer
+
+## Architecture
+
+The `GPv2VaultRelayer` contract is an important component used to protect user funds from malicious solvers. As previously mentioned, the `GPv2Settlement` contract allows using arbitrary on-chain liquidity through interactions (such as performing a swap on [Balancer V2](https://balancer.fi), or performing a Paraswap trade). If `Vault` and `ERC20` allowances were made directly to a `GPv2Settlement` contract, a malicious solver could drain user funds through the interaction mechanism. However, since these allowances are made to the `GPv2VaultRelayer` contract and interactions to the contract are strictly forbidden, malicious solvers have no direct access to user funds. The `GPv2Settlement` contract uses the `GPv2VaultRelayer` to withdraw user tokens only as part of the trade, which contains strong guarantees that the user's signed order parameters are respected.
+
+The `GPv2VaultRelayer` has access to user balances through 3 mechanisms:
+
+1. [Balancer External Balances](#balancer-external-balances)
+2. [Balancer Internal Balances](#balancer-internal-balances)
```
Comment (fleupold) [inline docs/cow-protocol/reference/contracts/core/vault_relayer.md]: These aren't really used in practice and should probably be explained with less priority.

File: docs/cow-protocol/reference/core/auctions/rewards.md [hunk 14]
```diff
@@ -0,0 +1,83 @@
...
+
+## Solver bidding strategies
+
+Apart from submitting their solutions, solvers must decide on a score to submit as a bid in the auction. In general, this score can be chosen freely by the solver (within the restriction $$0 < \textrm{score} < \textrm{successfulQuality}$$).
+
+Let $$p$$ the probability that a solution successfully executes, let $$SC$$ be the costs that a solver pays if only if a solution successfully executes, and $$C$$ the costs that a solver pays independently of whether a solution executes. For example, variable costs could include potential negative slippage of an executed solution, while fixed costs could include gas costs that arise even for a reverted solution.
+
+Ignoring capping, the winning solver's expected payoff is:
```
Comment (fleupold) [inline docs/cow-protocol/reference/core/auctions/rewards.md]: Maybe mention the easier to understand intuition that due to the cost from reverting settlements the expected quality should be discounted by the the revert probability in order to reach a score that leads to a positive payout in expectation. The cap is what makes everything messy.

File: docs/cow-protocol/reference/core/definitions.md [hunk 15]
```diff
@@ -0,0 +1,17 @@
+---
+sidebar_position: 1
+---
+
+# Definitions
+
+**Conditional Order**: An intent by a smart-contract to place `0..n` discrete orders under some programmatic logic. For example, a conditional order may only be valid if the price of an asset reaches a certain level.
```
Comment (fleupold) [inline docs/cow-protocol/reference/core/definitions.md]: Programmatic Orders?

**Review by fleupold — COMMENTED** [review]
Looks very good! Some comments inline

File: docs/cow-protocol/README.mdx (hunk 1 above)
Comment (avsavsavs) [reply docs/cow-protocol/README.mdx]: Included

File: docs/cow-protocol/concepts/batch-auctions/better-execution-prices.md (hunk 2 above)
Comment (avsavsavs) [reply docs/cow-protocol/concepts/batch-auctions/better-execution-prices.md]: Included " In essence, batch auctions turn the entire protocol into a giant barter economy where users can either trade directly against each other or directly against the best onchain liquidity source."

File: docs/cow-protocol/concepts/introduction/flow-of-an-order.md (hunk 4 above)
Comment (avsavsavs) [reply docs/cow-protocol/concepts/introduction/flow-of-an-order.md]: Included the comment

File: docs/cow-protocol/reference/contracts/core/README.mdx (hunk 12 above)
Comment (mfw78) [reply docs/cow-protocol/reference/contracts/core/README.mdx:34]: Added note linking signing schemes as the domain separators are covered extensively there.

File: docs/cow-protocol/reference/contracts/core/vault_relayer.md (hunk 13 above)
Comment (mfw78) [reply docs/cow-protocol/reference/contracts/core/vault_relayer.md]: Order adjusted to take this into account.

File: docs/cow-protocol/reference/core/auctions/rewards.md (hunk 14 above)
Comment (mfw78) [reply docs/cow-protocol/reference/core/auctions/rewards.md]: I do believe that this was reviewed and adjusted in PR #28 to take this into account.

**Pull request merged**